src/v/config/property.h

```diff
@@ -593,4 +596,27 @@ class enum_property : public property<T> {
     std::vector<T> _values;
 };
 
+class retention_property : public property<std::chrono::milliseconds> {
+public:
+    using property<std::chrono::milliseconds>::property;
+
+    void set_value(std::any v) override {
+        auto value = std::any_cast<std::chrono::milliseconds>(std::move(v));
+        if (value < std::chrono::milliseconds(0)) {
+            value = std::chrono::milliseconds(std::numeric_limits<int>::max());
```

Comment:
Is there a reason not to use `std::chrono::milliseconds::max()`?

2147483647ms is <25days.

Reply:
fixed